WeakAuras/GenericTrigger.lua

```diff
@@ -2509,7 +2509,9 @@ do
               -- a) We are already tracking it, only add the wathed spell ids
               -- b) We aren't tracking it yet add it
               if self.data[newEffectiveSpellId] then
-                tinsert(self.data[newEffectiveSpellId].watched, userSpellId)
+                if not tContains(self.data[newEffectiveSpellId].watched, userSpellId) then
+                  tinsert(self.data[newEffectiveSpellId].watched, userSpellId)
+                end
```

Comment:
I'm not really happy with this fix, though it does massively improve things - should this not be a set-like table instead of an array? But when I tried that, i got problems with scheduled timers, and when i tried to fix *that* i ran into a situation where the aura would vanish every time the base spell was replaced. So, I decided to instead push this up in the hopes @InfusOnWoW would understand the code better and decide on a better fix.

Reply:
I did at one point understand the code. Though it was a moment of clairvoyance.